### PR TITLE
feat: add cacheable directive to more queries

### PR DESCRIPTION
**Pull request by mzikherman**
Just a few more routes. All articles, and the A-Z route on /artists.

### CODE REVIEW

**Review by araujobarret — DISMISSED** [review]

**Review by damassi — DISMISSED** [review]
🔥

**Comment by mzikherman** [comment]
I pushed another commit to support `?nocache` query param always skipping the cache. This way, we have an escape hatch and can support any preview functionality that's needed.

File: src/System/Relay/createRelaySSREnvironment.ts [hunk 1]
```diff
@@ -99,7 +102,8 @@ export function createRelaySSREnvironment(config: Config = {}) {
       headers: req => {
         // Determine if the request is cacheable based on the opt-in `@cacheable` directive.
         // If it is, we don't want to send the user's access token even if they are logged in.
-        const isCacheable = isRequestCacheable(req)
+        const isCacheable =
```
Comment (mzikherman) [inline src/System/Relay/createRelaySSREnvironment.ts:105]: I'm down to iterate more on this, although I think this is fine for now.

At first I tried to wrap this condition in the `isRequestCacheable` method, but the `req` object we're generally passing around here, as these are all Relay middleware/helpers, is a `RelayRequest` (and not any Express/node request object) - which _doesnt_ have any URL/query params present.

File: src/System/Relay/middleware/__tests__/cacheHeaderMiddleware.jest.ts [hunk 2]
```diff
@@ -163,6 +163,33 @@ describe("cacheHeaderMiddleware", () => {
...
+          query ArtistQuery @cacheable {
+            artist(id: "andy-warhol) {
+              name
+            }
+          }
+        `,
+        }
+
+        jest.spyOn(window as any, "location", "get").mockImplementation(() => ({
+          pathname: "/artists?nocache=1",
```
Comment (damassi) [inline src/System/Relay/middleware/__tests__/cacheHeaderMiddleware.jest.ts:179]: Cool, so this just checks for existence. could be `nocache=true` too

Comment (mzikherman) [reply src/System/Relay/middleware/__tests__/cacheHeaderMiddleware.jest.ts:179]: Yep, anything truthy. `?nocache=` correctly doesn't count (that won't skip the cache). but `?nocache=blah|true|1`, etc. - anything else.

File: src/System/Relay/middleware/cacheHeaderMiddleware.ts [hunk 3]
```diff
@@ -34,6 +41,9 @@ export const shouldSkipCDNCache = (req, user, foundRoute) => {
   if (isLoggedIn) {
     return true
   }
+
+  // If none of the above conditions are met, we cache the request.
+  return false
```
Comment (mzikherman) [inline src/System/Relay/middleware/cacheHeaderMiddleware.ts:46]: This doesn't change anything as this method returned undefined (and so was fine when used in truth-y checks).

But - probably better to be explicit, esp as we should add some better typings to this file

**Review by damassi — APPROVED** [review]
🚀

**Pull request merged**